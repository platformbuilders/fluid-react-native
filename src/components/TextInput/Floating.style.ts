import styled from 'styled-components/native';
import { getTheme } from '@platformbuilders/theme-toolkit';
import TextInput from './index';

const success = getTheme('success.main');
const dangerMain = getTheme('danger.main');

const getColorByStatus = (props: any) => {
  const configByStatus: any = {
    default: {
      borderedColor: getTheme('text.light')(props),
      iconColor: getTheme('text.light')(props),
    },
    success: {
      borderedColor: success(props),
      iconColor: success(props),
    },
    error: {
      borderedColor: dangerMain(props),
      iconColor: dangerMain(props),
    },
    select: {
      borderedColor: getTheme('primary.main')(props),
      iconColor: getTheme('primary.main')(props),
    },
    focus: {
      borderedColor: getTheme('info.main')(props),
      iconColor: getTheme('info.main')(props),
    },
  };

  return configByStatus[props?.status || 'default'];
};

const getTextMain = getTheme('text.main');
const getBorderWidth = getTheme('borderWidth.xs');
const getFontSizeXs = getTheme('fontSizes.xs');
const getFontSizeLG = getTheme('fontSizes.lg');

const TextInputFloating = styled(TextInput).attrs((props: any) => {
  const fontSizeLg = getFontSizeLG(props) as number;
  const fontSizeXs = getFontSizeXs(props) as number;

  return {
    borderedHeight: 65,
    suppressAnimation: false,
    borderedRadius: 8,
    borderedWidth: getBorderWidth(props),
    borderedColor: getColorByStatus(props)?.borderedColor,
    iconColor: getColorByStatus(props)?.iconColor,
    withBottomline: false,
    fixedLabelVariant: 'animated',
    textStyle: {
      fontSize: fontSizeLg,
      color: getTextMain(props),
      paddingBottom: 5,
    },
    labelStyle: {
      lineHeight: 24,
      color: getTextMain(props),
    },
    iconSize: 18,
    animationValues: {
      upper: {
        top: 2,
        fontSize: fontSizeXs,
        opacity: 0.4,
      },
      lower: {
        top: 14,
        fontSize: fontSizeLg,
        opacity: 0.5,
      },
    },
  };
})``;

export default TextInputFloating;
