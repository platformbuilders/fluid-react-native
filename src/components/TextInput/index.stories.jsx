import { storiesOf } from '@storybook/react-native';
import textInputMarkdown from './textInput.md';

storiesOf('Text Input', module)
  .addParameters({
    readme: {
      content: textInputMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => {});
