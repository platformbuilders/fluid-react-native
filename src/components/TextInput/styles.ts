import { ComponentType } from 'react';
import { Animated, StyleSheet } from 'react-native';
import { moderateScale } from 'react-native-size-matters';
import styled from 'styled-components/native';
import { InputStatus } from '../../enums';
import { TextInputType } from '../../types';
import {
  getFontSize,
  getLineHeight,
  getTheme,
  ifStyle,
  scaledFontSize,
  switchStyle,
} from '../../utils/helpers';
import DefaultIcon from '../Icon';
import Typography from '../Typography';

const hasLeftIcon = ifStyle('hasLeftIcon');

type InputAreaWrapperProps = {
  rightIcon?: boolean;
  leftIcon?: boolean;
  multiline: boolean;
  padding?: number;
  inputRightPadding?: number;
  inputLeftPadding?: number;
};

type BorderedWrapperProps = {
  borderedBackgroundColor?: string;
  borderedHeight?: number;
  borderedColor?: string;
  borderedRadius?: number;
  error?: boolean;
  showBorderErrored?: boolean;
};

type InputBorderedAreaWrapperProps = {
  hasBottomLine?: boolean;
};

type InputBorderedColumnWrapperProps = {
  hasLeftIcon?: boolean;
  multiline?: boolean;
  padding?: number;
};

type FixedLabelAboveBorder = {
  style: Record<string, string>;
};

type FixedLabelProps = {
  hasLeftIcon?: boolean;
};

type BottomLineProps = {
  contrast: boolean;
  status: string;
};

const isMultiline = ifStyle('multiline');
const isCentered = ifStyle('centered');
const hasLabel = ifStyle('label');
const hasError = ifStyle('error');
const hasPadding = ifStyle('padding');
const hasInputRightPadding = ifStyle('inputRightPadding');
const hasInputLeftPadding = ifStyle('inputLeftPadding');
const hasBottomLine = ifStyle('hasBottomLine');
const isContrast = ifStyle('contrast');
const showIconErrored = ifStyle('showIconErrored');
const switchStatus = switchStyle('status');
const brandContrast = getTheme('brand.primary.contrast');
const brandPrimary = getTheme('brand.primary.main');
const minimumSpacing = getTheme('spacing.xs');
const smallSpacing = getTheme('spacing.sm');
const extraLargeSpacing = getTheme('spacing.xl');
const success = getTheme('success.main');
const textColor = getTheme('text.main');
const dangerMain = getTheme('danger.main');
const disabled = getTheme('brand.primary.main');
const inputColor = (props: TextInputType | BottomLineProps): any =>
  switchStatus({
    [InputStatus.Success]: success(props),
    [InputStatus.Danger]: dangerMain(props),
    [InputStatus.Default]: isContrast(
      brandContrast(props),
      textColor(props),
    )(props),
    [InputStatus.Disabled]: disabled(props),
  });

export const LABEL_UPPER_STYLE = {
  top: -10,
  fontSize: scaledFontSize(14),
};

export const LABEL_LOWER_STYLE = {
  top: 8,
  fontSize: scaledFontSize(18),
};

type WrapperProps = {
  multiline?: boolean;
  style?: any;
};

export const Wrapper = styled.View<WrapperProps>`
  justify-content: ${hasLabel('flex-end', 'flex-start')};
  padding-top: ${hasLabel(smallSpacing, 0)}px;
  position: relative;
`;

export const BorderedWrapper = styled.View<BorderedWrapperProps>`
  ${({
    borderedBackgroundColor,
    borderedColor,
    borderedHeight,
    borderedRadius,
    error,
    showBorderErrored,
    ...rest
  }: BorderedWrapperProps) => {
    const borderedStyle = `
      justify-content: center;
      border: 1px solid ${
        error && showBorderErrored
          ? dangerMain(rest)
          : borderedColor || brandPrimary(rest)
      };
      background-color: ${borderedBackgroundColor || 'transparent'};
      height: ${borderedHeight}px;
      border-radius: ${borderedRadius}px;
      padding: ${smallSpacing(rest)}px;
    `;

    return `
    border: 0;
    ${borderedHeight ? borderedStyle : ''}
  `;
  }}
`;

export const InputBorderedAreaWrapper = styled.View<InputBorderedAreaWrapperProps>`
  flex-direction: row;
  align-items: center;
  width: 100%;
  height: ${hasBottomLine('100%', 'auto')};
`;

export const InputBorderedColumnWrapper = styled.View<InputBorderedColumnWrapperProps>`
  flex-direction: column;
  padding: 0 ${minimumSpacing}px;
  width: ${hasLeftIcon('86%', '92%')};
  margin-left: ${hasLeftIcon(minimumSpacing, 0)}px;
`;

export const FixedLabelAboveBorder = styled(Typography)<FixedLabelAboveBorder>`
  color: ${textColor};
  position: absolute;
  top: -${extraLargeSpacing}px;
  left: 0;
`;

export const FixedLabel = styled(Typography)<FixedLabelProps>`
  color: ${brandPrimary};
  margin-bottom: ${minimumSpacing}px;
`;

export const InputAreaWrapper = styled.View<InputAreaWrapperProps>`
  padding-top: ${hasPadding(
    ({ padding }: InputAreaWrapperProps) => padding,
    minimumSpacing,
  )}px;
  padding-right: ${hasInputRightPadding(
    ({ inputRightPadding }: InputAreaWrapperProps) => inputRightPadding,
    hasPadding(({ padding }: InputAreaWrapperProps) => padding, minimumSpacing),
  )}px;
  padding-left: ${hasInputLeftPadding(
    ({ inputLeftPadding }: InputAreaWrapperProps) => inputLeftPadding,
    hasPadding(({ padding }: InputAreaWrapperProps) => padding, minimumSpacing),
  )}px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  overflow: hidden;
`;

export const TextLabel = styled.Text<any>`
  font-size: ${getFontSize}px;
  line-height: ${getLineHeight}px;
  position: absolute;
  color: ${inputColor};
  top: ${LABEL_LOWER_STYLE.top}px;
`;

export const Label = Animated.createAnimatedComponent<ComponentType<any>>(
  TextLabel,
);

export const TextInput = styled.TextInput.attrs((props: TextInputType) => ({
  accessibilityLabel: props.accessibilityLabel || props.accessibility,
  testID: props.testID || props.id,
  textAlign: isCentered('center', 'left')(props),
  placeholderTextColor: props.placeholderTextColor
    ? props.placeholderTextColor
    : '#72727260',
}))<TextInputType>`
  padding: 0;
  flex-grow: 1;
  border-width: 0;
  min-height: ${moderateScale(24)}px;
  color: ${inputColor};
  margin-top: ${isMultiline(smallSpacing, 0)}px;
  font-size: ${getFontSize}px;
  width: 100%;
`;

export const BottomLine = styled.View<BottomLineProps>`
  height: ${StyleSheet.hairlineWidth}px;
  background-color: ${inputColor};
`;

type IconProps = {
  contrast: boolean;
  error: boolean;
  iconColor?: string;
  showIconErrored?: boolean;
};

const defaultIconColor = (props: IconProps) =>
  props.iconColor ||
  isContrast(brandContrast(props), brandPrimary(props))(props);

const erroredIconColor = (props: IconProps) => dangerMain(props);

export const Icon = styled(DefaultIcon).attrs((props: IconProps) => ({
  color: showIconErrored(
    hasError(erroredIconColor(props), defaultIconColor(props))(props),
    defaultIconColor(props),
  )(props),
}))<IconProps>``;
