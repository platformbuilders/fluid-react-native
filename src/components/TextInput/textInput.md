
# Text Input âŒ¨ 

## Example
![Text Input](https://user-images.githubusercontent.com/44801113/81105431-46d3d280-8eea-11ea-81cc-ee1cf1351b39.jpeg)



## Source Code
```js
import React from 'react';
import { TextInput } from '@platformbuilders/react-native-ui';

const Home: React.FC = () => {
  return (
      <TextInput
        large
        id="products_searc"
        accessibility="Product Search"
        placeholder="Product Search "
        contrast
        centered
        multiline
        iconTouchableEnabled
        iconName="barcode"
        iconSize={40}
        label="Products"
        error="has a error"
      />
  );
}

export default Home;
```

## Props 

| Name  | Required | Type | Default Value | Description 
| ------------- | ------------- | ------------- |------------- |------------- |
| id | true | string | ' '  |   |
| accessibility | true | string | ' '  |   |
| accessibilityLabel | true | string | ' '  |   |
| testID | true | string | ' '  |   |
| large | false | boolean | false  | size |
| contrast | false | boolean | false  | color  |
| inputRef | false | any |   | ref of input  |
| centered | false | boolean | false  | center input |
| withoutBottomline | false | boolean | false  | hide border bottom |
| multiline | false | boolean | false  |  |
| keyboardType | false | string | default  |  |
| keyboardType | false | string | default  |  |
| iconName | false | string | ' '  |  name of icon right |
| iconTouchableEnabled | false | boolean | false  |  icon clickable |
| iconSize | false | number | 20  |  icon size |
| status | false | string | DEFAULT  | options: SUCCES, FAILURE, DISABLED, DEFAULT |
| maskType | false | string | null  | options: credit-card, cpf, cnpj, zip-code, only-numbers, money, cel-phone, datetime, custom |
| label | false | string | ' '  |   |
| value | false | any | ' '  |  value of input |
| placeholder | false | string | ' '  |  placeholder of input |
| error | false | string | ' '  |  error message in bottom |
| style | false | object | {} |  style input |
| textStyle | false | object | {} |   |
| onBlur | false | function | (): any => {} |  handle input  |
| onFocus | false | function | (): any => {} |  handle input  |
| onChangeText | false | function | (): any => {} |  handle input  |
| onPressIcon | false | function | (): any => {} |  handle press icon  |








