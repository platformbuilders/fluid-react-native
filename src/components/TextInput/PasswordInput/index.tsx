import React, { useState, FC, useCallback } from 'react';
import { TextInputType } from '../../../types';
import TextInput from '../index';

const PasswordInput: FC<TextInputType> = (props) => {
  const [hidePassword, setHidePassword] = useState<boolean>(true);
  const hitSlop = {
    left: 40,
    right: 40,
    top: 40,
    bottom: 40,
  };

  const onPressShowPassword = useCallback((): void => {
    setHidePassword(!hidePassword);
  }, [hidePassword]);

  return (
    <TextInput
      secureTextEntry={hidePassword}
      iconName={hidePassword ? 'eye' : 'eye-off'}
      iconTouchableEnabled
      onPressIcon={onPressShowPassword}
      iconHitSlop={hitSlop}
      {...props}
    />
  );
};

export default PasswordInput;
