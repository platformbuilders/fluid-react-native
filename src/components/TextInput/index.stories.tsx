import React from 'react';
import { View } from 'react-native';
import type { Meta, StoryObj } from '@storybook/react';
import TextInputFloating from './Floating.style';
import TextInput from '.';

const spacing = { marginTop: 16 };

const Render = ({ ...args }) => {
  return (
    <View style={{ width: 420 }}>
      <TextInputFloating
        {...args}
        id="textInput"
        accessibility="Este componente TextInputFloating"
        style={spacing}
      />
      <TextInputFloating
        {...args}
        id="textInput"
        accessibility="Este componente TextInputFloating"
        style={spacing}
      />
      <TextInputFloating
        {...args}
        id="textInput"
        accessibility="Este componente TextInputFloating"
        style={spacing}
        status="error"
      />
      <TextInputFloating
        {...args}
        id="textInput"
        accessibility="Este componente TextInputFloating"
        style={spacing}
        status="success"
      />
      <TextInputFloating
        {...args}
        id="textInput"
        accessibility="Este componente TextInputFloating"
        style={spacing}
        status="selected"
      />
      <TextInputFloating
        {...args}
        id="textInput"
        accessibility="Este componente TextInputFloating"
        style={spacing}
        status="focus"
      />
    </View>
  );
};

const meta: Meta<typeof TextInput> = {
  title: 'TextInput',
  component: Render,
  args: {
    id: 'textInput',
    label: 'Default',
    value: '',
  },
  parameters: {
    notes: 'Este componente é utilizado para renderizar um TextInput.',
  },
};

export default meta;
type Story = StoryObj<typeof TextInput>;

export const Floating: Story = {
  args: {
    id: 'textInput',
    label: 'Input value here',
    iconName: 'flag',
    isFloating: true,
  },
  parameters: {
    notes: 'Este componente é utilizado para renderizar um TextInputFloating.',
  },
};
