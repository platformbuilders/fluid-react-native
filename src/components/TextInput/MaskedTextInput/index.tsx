import React, { FC } from 'react';
import { MaskedTextInputType, InputStatus } from '../../../types';
import { TextInput } from './styles';

const MaskedTextInput: FC<MaskedTextInputType> = ({
  maskType,
  inputRef,
  id,
  accessibility,
  contrast = false,
  multiline = false,
  status = InputStatus.Default,
  ...props
}) => (
  <TextInput
    {...props}
    id={id || accessibility}
    status={status}
    accessibility={accessibility}
    testID={id || accessibility}
    accessibilityLabel={accessibility}
    ref={inputRef}
    contrast={contrast}
    multiline={multiline}
    type={maskType}
    underlineColorAndroid="transparent"
  />
);

export default MaskedTextInput;
