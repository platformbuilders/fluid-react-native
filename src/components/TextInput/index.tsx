import React, { FC, useCallback, useEffect, useRef, useState } from 'react';
import { isEmpty } from 'lodash';
import { Animated } from 'react-native';
import { InputStatus, TextInputType, TypographyVariants } from '../../types';
import { usePrevious } from '../../utils/hooks';

import FormError from '../FormError';
import MaskedTextInput from './MaskedTextInput';
import {
  BorderedWrapper,
  BottomLine,
  FixedLabel,
  Icon,
  InputAreaWrapper,
  InputBorderedAreaWrapper,
  InputBorderedColumnWrapper,
  LABEL_LOWER_STYLE,
  LABEL_UPPER_STYLE,
  Label,
  Wrapper,
} from './styles';

const TextInput: FC<TextInputType> = ({
  id,
  accessibility,
  accessibilityLabel,
  testID,
  large = false,
  contrast = false,
  centered = false,
  withBottomline = true,
  multiline = false,
  autoFocus = false,
  allowFontScaling = false,
  keyboardType = 'default',
  iconSize = 20,
  iconTouchableEnabled = false,
  status = InputStatus.Default,
  maskType = null,
  iconNameBordered = '',
  iconName,
  label = '',
  value = '',
  placeholder = '',
  error = '',
  style = {},
  textStyle = {},
  labelStyle = {},
  iconHitSlop = {},
  inputRef = useRef(null),
  onBlur = (): any => {},
  onFocus = (): any => {},
  onChangeText = (): any => {},
  onPressIcon = (): any => {},
  onRightIconPress = (): any => {},
  inputLeftPadding,
  inputRightPadding,
  rightIconName,
  leftIconName,
  leftIconColor,
  iconColor,
  inputPadding,
  borderedBackgroundColor,
  borderedHeight,
  borderedColor,
  borderedRadius,
  iconType = 'material',
  fixedLabelVariant = 'xs',
  suppressAnimation = false,
  ...rest
  // eslint-disable-next-line sonarjs/cognitive-complexity
}) => {
  const animationInitialValues = {
    top: suppressAnimation ? LABEL_UPPER_STYLE.top : LABEL_LOWER_STYLE.top,
    fontSize: suppressAnimation
      ? LABEL_UPPER_STYLE.fontSize
      : LABEL_LOWER_STYLE.fontSize,
  };

  const [labelAnimatedStyle] = useState({
    top: new Animated.Value(animationInitialValues.top),
    fontSize: new Animated.Value(animationInitialValues.fontSize),
  });
  const [isPlaceholder, setIsPlaceHolder] = useState(
    suppressAnimation ? false : true,
  );
  const previousValue = usePrevious<string>(value || '');
  const labelVariant: TypographyVariants = large ? 'md' : 'xs';
  const textVariant: TypographyVariants = large ? 'lg' : 'md';
  const placeholderVariant: TypographyVariants = large ? 'lg' : 'md';

  const animateComponent = useCallback(
    (updatedLabelStyle: any): void => {
      const animations = Object.keys(updatedLabelStyle).map((animationProp) =>
        Animated.timing(labelAnimatedStyle[animationProp], {
          toValue: updatedLabelStyle[animationProp],
          duration: 200,
          useNativeDriver: false,
        }),
      );
      Animated.parallel(animations).start();
    },
    [labelAnimatedStyle],
  );

  const animationUp = (): void => {
    animateComponent(LABEL_UPPER_STYLE);
  };

  const animationDown = (): void => {
    animateComponent(LABEL_LOWER_STYLE);
  };

  const handleOnFocus = (event: any): void => {
    if (isPlaceholder && !suppressAnimation) {
      setIsPlaceHolder(false);
      animationUp();
    }
    onFocus(event);
  };

  const handleOnBlur = (event: any): void => {
    const isEmptyLabel = label === '';
    if (!value && !isEmptyLabel && !suppressAnimation) {
      setIsPlaceHolder(true);
      animationDown();
    }
    if (isEmptyLabel) {
      setIsPlaceHolder(false);
    }
    onBlur(event);
  };

  const renderTextInput = (inputStatus: string): JSX.Element => {
    const renderPlaceholder = !value && !isPlaceholder ? placeholder : '';

    const textInputProps = {
      id,
      accessibility,
      accessibilityLabel,
      testID,
      large,
      variant: textVariant,
      centered,
      contrast,
      withBottomline,
      multiline,
      value,
      keyboardType,
      onChangeText,
      allowFontScaling,
      isPlaceholder,
      status: inputStatus,
      placeholder: renderPlaceholder,
      style: textStyle,
      onBlur: handleOnBlur,
      onFocus: handleOnFocus,
      autoFocus,
      underlineColorAndroid: 'transparent',
      ...rest,
    };

    return (
      <MaskedTextInput
        inputRef={inputRef}
        maskType={maskType || 'no-mask'}
        accessibilityLabel={accessibilityLabel || accessibility}
        testID={testID || id || accessibility}
        {...textInputProps}
      />
    );
  };

  const setAnimation = () => {
    const wasEmpty = previousValue?.length === 0;
    if (value && value.length && wasEmpty && !suppressAnimation) {
      animationUp();
    }
    if (label === '') {
      setIsPlaceHolder(false);
    }
  };

  useEffect(() => {
    setAnimation();
  }, [value, previousValue]);

  const hasError = !isEmpty(error);

  const icon = iconName;
  const iconBordered = iconNameBordered;
  const renderStatus = hasError ? InputStatus.Danger : status;

  const renderIcon = (iconProp: string, isLeft?: boolean) => (
    <Icon
      type={iconType}
      id={`id_${iconProp}`}
      accessibility={`icon_${accessibility}`}
      size={iconSize}
      name={iconProp || ''}
      contrast={contrast}
      error={hasError}
      touchable={iconTouchableEnabled}
      onPress={!!rightIconName ? onRightIconPress : onPressIcon}
      hitSlop={iconHitSlop}
      iconColor={isLeft ? leftIconColor : iconColor}
    />
  );

  return (
    <Wrapper style={style} multiline={multiline}>
      <FormError
        id={id || accessibility}
        accessibility={accessibility}
        centered={centered}
        error={error}
        large={large}
      >
        <BorderedWrapper
          borderedBackgroundColor={borderedBackgroundColor}
          borderedHeight={borderedHeight}
          borderedColor={borderedColor}
          borderedRadius={borderedRadius}
          error={hasError}
        >
          {!centered && !borderedHeight && (
            <Label
              status={status}
              contrast={contrast}
              style={[labelAnimatedStyle, labelStyle]}
              variant={isPlaceholder ? placeholderVariant : labelVariant}
              testID={`error_${id || accessibility}`}
              accessibilityLabel={`Erro ${accessibility}`}
            >
              {label}
            </Label>
          )}
          {borderedHeight ? (
            <InputBorderedAreaWrapper>
              {!isEmpty(iconBordered) && renderIcon(iconBordered, true)}
              <InputBorderedColumnWrapper
                hasLeftIcon={!isEmpty(iconBordered)}
                multiline={multiline}
                padding={inputPadding}
              >
                <FixedLabel
                  hasLeftIcon={!isEmpty(iconBordered)}
                  variant={fixedLabelVariant}
                >
                  {label}
                </FixedLabel>
                {renderTextInput(renderStatus)}
              </InputBorderedColumnWrapper>
              {!isEmpty(icon) && renderIcon(icon || '')}
            </InputBorderedAreaWrapper>
          ) : (
            <InputAreaWrapper
              multiline={multiline}
              padding={inputPadding}
              rightIcon={!!rightIconName}
              inputLeftPadding={inputLeftPadding}
              inputRightPadding={inputRightPadding}
            >
              {borderedHeight && <FixedLabel>{label}</FixedLabel>}
              {!!leftIconName && renderIcon(leftIconName)}
              {renderTextInput(renderStatus)}
              {!!rightIconName && renderIcon(rightIconName)}
              {!leftIconName &&
                !rightIconName &&
                !isEmpty(icon) &&
                renderIcon(icon as string)}
            </InputAreaWrapper>
          )}
          {withBottomline && <BottomLine status={status} contrast={contrast} />}
        </BorderedWrapper>
      </FormError>
    </Wrapper>
  );
};

export default TextInput;
