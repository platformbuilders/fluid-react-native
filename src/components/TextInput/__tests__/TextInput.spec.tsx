import 'jest';
import React, { useState, FC } from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { ThemeProvider } from 'styled-components/native';
import TextInput from '..';
import { theme } from '../../../test/helpers';

describe('<TextInput />', () => {
  it('should render textinput', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <TextInput id="test" accessibility="" />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render input and trigger state change, blur, focus and onPressIcon', () => {
    const onBlurFN = jest.fn();
    const onFocusFN = jest.fn();
    const Component: FC = (): JSX.Element => {
      const [value, setValue] = useState('');

      return (
        <ThemeProvider theme={theme}>
          <TextInput
            id="testing_textInput"
            accessibility=""
            value={value}
            onChangeText={setValue}
            onBlur={onBlurFN}
            onFocus={onFocusFN}
            iconName="access-point"
          />
        </ThemeProvider>
      );
    };

    const { getByTestId } = render(<Component />);

    const component = getByTestId('testing_textInput');

    fireEvent.changeText(component, 'Value changed');
    fireEvent(component, 'blur');
    fireEvent(component, 'focus');

    expect(onBlurFN).toHaveBeenCalled();
    expect(onFocusFN).toHaveBeenCalled();
    expect(component.props.value).toBe('Value changed');
  });
});
