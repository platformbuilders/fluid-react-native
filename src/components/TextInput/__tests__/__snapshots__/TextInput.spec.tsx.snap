// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render textinput 1`] = `
<View
  multiline={false}
  style={
    Array [
      Object {
        "justifyContent": "flex-start",
        "paddingTop": 0,
        "position": "relative",
      },
      Object {},
    ]
  }
>
  <Text
    accessibilityLabel="Erro "
    contrast={false}
    status="DEFAULT"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 31,
          "lineHeight": 43,
          "position": "absolute",
          "top": 20,
        },
        Object {
          "fontSize": 35,
          "top": 20,
        },
      ]
    }
    testID="error_test"
    variant="body"
  >
    
  </Text>
  <View
    multiline={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": 18.857142857142858,
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      accessibility=""
      accessibilityLabel=""
      allowFontScaling={false}
      withoutBottomline={true}
      centered={false}
      contrast={false}
      id="test"
      isPlaceholder={true}
      keyboardType="default"
      large={false}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#72727260"
      rejectResponderTermination={true}
      status="DEFAULT"
      style={
        Array [
          Object {
            "borderWidth": 0,
            "color": "#212121",
            "flexGrow": 1,
            "fontSize": 33,
            "lineHeight": 43,
            "marginTop": 0,
            "minHeight": 37.714285714285715,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          Object {},
        ]
      }
      testID="test"
      textAlign="left"
      underlineColorAndroid="transparent"
      value=""
      variant="headline"
    />
  </View>
  <View
    contrast={false}
    status="DEFAULT"
    style={
      Array [
        Object {
          "backgroundColor": "#212121",
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;
